src/compiler/factory/node_factory.rs

```diff
@@ -240,3 +273,57 @@ impl NodeFactory {
 pub fn create_node_factory() -> NodeFactory {
     NodeFactory {}
 }
+
+// lazy_static! {
+//     static ref base_factory_static: BaseNodeFactoryConcrete = create_base_node_factory();
```

Comment:
Because I need the interface to `BaseNodeFactory` to take immutable `&self` references (because `ParserType` implements `BaseNodeFactory` and needed to pass immutable rather than mutable references to itself into the node factory methods), `BaseNodeFactoryConcrete` which lazily initializes its "node constructor" state has to use `RefCell`'s under the hood which don't meet the `Sync` or whatever requirements for `lazy_static!`

So similar to what I've done in a couple other places, rather than exposing a static singleton I expose a "creator helper" (in this case `create_base_node_factory()`)

The same thing "infected" `BaseNodeFactorySynthetic` below I believe so similarly exposed a `get_synthetic_factory()` "creator helper" rather than a `synthetic_factory` static singleton